src/main/java/uk/ac/ebi/atlas/solr/analytics/AnalyticsSearchService.java

```diff
@@ -0,0 +1,99 @@
+package uk.ac.ebi.atlas.solr.analytics;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.jayway.jsonpath.JsonPath;
+import net.minidev.json.JSONArray;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.search.SemanticQuery;
+import uk.ac.ebi.atlas.species.SpeciesFinder;
+
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class AnalyticsSearchService implements SpeciesFinder {
+    private final MiscellaneousAnalyticsSearchDao miscellaneousAnalyticsSearchDao;
+
+    public AnalyticsSearchService(MiscellaneousAnalyticsSearchDao miscellaneousAnalyticsSearchDao) {
+        this.miscellaneousAnalyticsSearchDao = miscellaneousAnalyticsSearchDao;
+    }
+
+    public ImmutableSet<String> fetchExperimentTypes(String bioentityIdentifier) {
+        return fetchExperimentTypes(SemanticQuery.create(bioentityIdentifier), SemanticQuery.create(), "");
+    }
+
+    public static ImmutableSet<String> readBuckets(String response) {
+        List<Map<String, Object>> res = JsonPath.read(response, "$..buckets[*]");
+
+        ImmutableSet.Builder<String> b = ImmutableSet.builder();
+        for (Map<String, Object> m: res) {
```

Comment:
Can use the latest `forEach with lambda expressions` if possible, please.

Reply:
I didn’t do it so that I’d modify as little code as possible but it’s not much effort, so I did it after your suggestion.